### PR TITLE
Adds basic support for AtlanticElectricalHeaterWithAdjustableTemperatureSetpoint

### LINKED ISSUE #304: Add support for AtlanticElectricalHeaterWithAdjustableTemperatureSetpoint (io:AtlanticElectricalHeaterWithAdjustableTemperatureSetpointIOComponent)
**Device information**
Sauter Ipala elelectrical heater (1000W and 1500W) not recognized as a Climate entity. 
Would like to be able to set the target temperature and the operating mode (BASIC, PROG, VACATION, OFF).
Temperature, Contact and Occupancy sensors are discovered.

**Device details**
2020-10-25 10:23:48 DEBUG (MainThread) [custom_components.tahoma] Unsupported TaHoma device detected (io:AtlanticElectricalHeaterWithAdjustableTemperatureSetpointIOComponent - HeatingSystem - AtlanticElectricalHeaterWithAdjustableTemperatureSetpoint)

```json
{
	"commands": [{
		"commandName": "cancelHeatingLevel",
		"nparams": 1
	}, {
		"commandName": "delayedStopIdentify",
		"nparams": 1
	}, {
		"commandName": "getName",
		"nparams": 0
	}, {
		"commandName": "identify",
		"nparams": 0
	}, {
		"commandName": "noPersonInside",
		"nparams": 0
	}, {
		"commandName": "off",
		"nparams": 0
	}, {
		"commandName": "personInside",
		"nparams": 0
	}, {
		"commandName": "refreshComfortTemperature",
		"nparams": 0
	}, {
		"commandName": "refreshDateTime",
		"nparams": 0
	}, {
		"commandName": "refreshDerogatedTargetTemperature",
		"nparams": 0
	}, {
		"commandName": "refreshEcoTemperature",
		"nparams": 0
	}, {
		"commandName": "refreshHeatingLevel",
		"nparams": 0
	}, {
		"commandName": "refreshIdentifier",
		"nparams": 0
	}, {
		"commandName": "refreshManufacturerName",
		"nparams": 0
	}, {
		"commandName": "refreshMaximumHeatingTargetTemperature",
		"nparams": 0
	}, {
		"commandName": "refreshMaximumTargetTemperature",
		"nparams": 0
	}, {
		"commandName": "refreshOperatingMode",
		"nparams": 0
	}, {
		"commandName": "refreshTargetTemperature",
		"nparams": 0
	}, {
		"commandName": "refreshTemperature",
		"nparams": 0
	}, {
		"commandName": "refreshTimeProgram",
		"nparams": 0
	}, {
		"commandName": "setComfortTemperature",
		"nparams": 1
	}, {
		"commandName": "setDateTime",
		"nparams": 1
	}, {
		"commandName": "setDerogatedTargetTemperature",
		"nparams": 1
	}, {
		"commandName": "setEcoTemperature",
		"nparams": 1
	}, {
		"commandName": "setHeatingLevel",
		"nparams": 1
	}, {
		"commandName": "setHeatingLevelWithTimer",
		"nparams": 2
	}, {
		"commandName": "setHolidays",
		"nparams": 1
	}, {
		"commandName": "setHolidaysTargetTemperature",
		"nparams": 1
	}, {
		"commandName": "setName",
		"nparams": 1
	}, {
		"commandName": "setOccupancyActivation",
		"nparams": 1
	}, {
		"commandName": "setOccupancy",
		"nparams": 1
	}, {
		"commandName": "setOpenWindowDetectionActivation",
		"nparams": 1
	}, {
		"commandName": "setOperatingMode",
		"nparams": 1
	}, {
		"commandName": "setPreviousTargetTemperature",
		"nparams": 1
	}, {
		"commandName": "setSchedulingType",
		"nparams": 1
	}, {
		"commandName": "setTargetTemperature",
		"nparams": 1
	}, {
		"commandName": "setTimeProgram",
		"nparams": 1
	}, {
		"commandName": "startIdentify",
		"nparams": 0
	}, {
		"commandName": "stopIdentify",
		"nparams": 0
	}, {
		"commandName": "wink",
		"nparams": 1
	}, {
		"commandName": "pairOneWayController",
		"nparams": 2
	}, {
		"commandName": "refreshAutoProgram",
		"nparams": 0
	}, {
		"commandName": "refreshControllerAddress",
		"nparams": 0
	}, {
		"commandName": "refreshCumulatedLowering",
		"nparams": 0
	}, {
		"commandName": "refreshCurrentWorkingRate",
		"nparams": 0
	}, {
		"commandName": "refreshDeletionCancelation",
		"nparams": 0
	}, {
		"commandName": "refreshEffectiveTemperatureSetpoint",
		"nparams": 0
	}, {
		"commandName": "refreshLocalLeadTime",
		"nparams": 0
	}, {
		"commandName": "refreshModel",
		"nparams": 0
	}, {
		"commandName": "refreshNativeFunctionalLevel",
		"nparams": 0
	}, {
		"commandName": "refreshPeakNotice",
		"nparams": 0
	}, {
		"commandName": "refreshPeakWarning",
		"nparams": 0
	}, {
		"commandName": "refreshPowerAndTension",
		"nparams": 0
	}, {
		"commandName": "refreshRoomDeletionThreshold",
		"nparams": 0
	}, {
		"commandName": "refreshSetpointLoweringTemperatureInProgMode",
		"nparams": 0
	}, {
		"commandName": "refreshSynchronisationRequest",
		"nparams": 0
	}, {
		"commandName": "refreshTemperatureProbeCalibrationOffset",
		"nparams": 0
	}, {
		"commandName": "setCommunicationTest",
		"nparams": 1
	}, {
		"commandName": "setDeletionCancelation",
		"nparams": 1
	}, {
		"commandName": "setExpectedPresence",
		"nparams": 1
	}, {
		"commandName": "setHeatingLevelForTrigger",
		"nparams": 1
	}, {
		"commandName": "setNativeLevelsList",
		"nparams": 1
	}, {
		"commandName": "setPeakNotice",
		"nparams": 1
	}, {
		"commandName": "setPeakWarning",
		"nparams": 1
	}, {
		"commandName": "setRoomDeletionThreshold",
		"nparams": 1
	}, {
		"commandName": "setSetpointLoweringTemperatureInProgMode",
		"nparams": 1
	}, {
		"commandName": "setTemperatureProbeCalibrationOffset",
		"nparams": 1
	}, {
		"commandName": "setTwinningExit",
		"nparams": 1
	}, {
		"commandName": "unpairAllOneWayControllers",
		"nparams": 0
	}, {
		"commandName": "unpairOneWayController",
		"nparams": 2
	}],
	"states": [{
		"type": "ContinuousState",
		"qualifiedName": "core:ComfortRoomTemperatureState"
	}, {
		"type": "DataState",
		"qualifiedName": "core:DateTimeState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "core:DerogatedTargetTemperatureState"
	}, {
		"values": ["good", "low", "normal", "verylow"],
		"type": "DiscreteState",
		"qualifiedName": "core:DiscreteRSSILevelState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "core:EcoRoomTemperatureState"
	}, {
		"values": ["off", "on"],
		"type": "DiscreteState",
		"qualifiedName": "core:HolidaysModeState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "core:HolidaysTargetTemperatureState"
	}, {
		"type": "DataState",
		"qualifiedName": "core:IdentifierState"
	}, {
		"type": "DataState",
		"qualifiedName": "core:ManufacturerNameState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "core:MaximumHeatingTargetTemperatureState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "core:MaximumTargetTemperatureState"
	}, {
		"type": "DataState",
		"qualifiedName": "core:NameState"
	}, {
		"values": ["active", "inactive"],
		"type": "DiscreteState",
		"qualifiedName": "core:OccupancyActivationState"
	}, {
		"values": ["off", "on"],
		"type": "DiscreteState",
		"qualifiedName": "core:OnOffState"
	}, {
		"values": ["active", "inactive"],
		"type": "DiscreteState",
		"qualifiedName": "core:OpenWindowDetectionActivationState"
	}, {
		"values": ["antifreeze", "auto", "away", "eco", "frostprotection", "manual", "max", "normal", "off", "on", "prog", "program"],
		"type": "DiscreteState",
		"qualifiedName": "core:OperatingModeState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "core:PreviousTargetTemperatureState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "core:PriorityLockTimerState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "core:RSSILevelState"
	}, {
		"values": ["increase", "none", "standby"],
		"type": "DiscreteState",
		"qualifiedName": "core:RegulationModeState"
	}, {
		"values": ["available", "unavailable"],
		"type": "DiscreteState",
		"qualifiedName": "core:StatusState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "core:TargetTemperatureState"
	}, {
		"type": "DataState",
		"qualifiedName": "core:TimeProgramState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "core:VersionState"
	}, {
		"type": "DataState",
		"qualifiedName": "io:AutoProgramState"
	}, {
		"type": "DataState",
		"qualifiedName": "io:ControllerAddressState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "io:CumulatedLoweringState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "io:CurrentWorkingRateState"
	}, {
		"values": ["deletion cancelation", "no deletion cancelation"],
		"type": "DiscreteState",
		"qualifiedName": "io:DeletionCancelationState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "io:EffectiveTemperatureSetpointState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "io:ExpectedPresenceState"
	}, {
		"values": ["external", "internal"],
		"type": "DiscreteState",
		"qualifiedName": "io:InternalExternalSchedulingTypeState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "io:LocalLeadTimeState"
	}, {
		"values": ["boost", "comfort", "comfort-1", "comfort-2", "eco", "frostprotection", "off", "secured"],
		"type": "DiscreteState",
		"qualifiedName": "io:MaximumHeatingLevelState"
	}, {
		"type": "DataState",
		"qualifiedName": "io:ModelState"
	}, {
		"values": ["base", "medium", "top"],
		"type": "DiscreteState",
		"qualifiedName": "io:NativeFunctionalLevelState"
	}, {
		"values": ["long peak", "no peak", "short peak"],
		"type": "DiscreteState",
		"qualifiedName": "io:PeakNoticeState"
	}, {
		"values": ["long peak warning", "no warning", "short peak warning"],
		"type": "DiscreteState",
		"qualifiedName": "io:PeakWarningState"
	}, {
		"type": "DataState",
		"qualifiedName": "io:PowerState"
	}, {
		"values": ["comfortLevel1", "comfortLevel2", "comfortLevel3", "comfortLevel4", "environmentProtection", "humanProtection", "userLevel1", "userLevel2"],
		"type": "DiscreteState",
		"qualifiedName": "io:PriorityLockLevelState"
	}, {
		"values": ["LSC", "SAAC", "SFC", "UPS", "externalGateway", "localUser", "myself", "rain", "security", "temperature", "timer", "user", "wind"],
		"type": "DiscreteState",
		"qualifiedName": "io:PriorityLockOriginatorState"
	}, {
		"type": "DataState",
		"qualifiedName": "io:RoomDeletionThresholdState"
	}, {
		"values": ["kept", "lost"],
		"type": "DiscreteState",
		"qualifiedName": "io:RunningState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "io:SetpointLoweringTemperatureInProgModeState"
	}, {
		"values": ["boost", "comfort", "comfort-1", "comfort-2", "eco", "frostprotection", "off", "secured"],
		"type": "DiscreteState",
		"qualifiedName": "io:TargetHeatingLevelState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "io:TemperatureProbeCalibrationOffsetState"
	}, {
		"type": "DataState",
		"qualifiedName": "io:TensionState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "io:TimerForTransitoryStateState"
	}, {
		"type": "ContinuousState",
		"qualifiedName": "io:UptimeState"
	}],
	"dataProperties": [{
		"value": "500",
		"qualifiedName": "core:identifyInterval"
	}],
	"widgetName": "AtlanticElectricalHeaterWithAdjustableTemperatureSetpoint",
	"uiClass": "HeatingSystem",
	"uiClassifiers": ["emitter"],
	"qualifiedName": "io:AtlanticElectricalHeaterWithAdjustableTemperatureSetpointIOComponent",
	"type": "ACTUATOR"
}
```

Related issues: https://github.com/iMicknl/ha-tahoma/issues/339

### PR DESCRIPTION
**Pull request by iMicknl**
Fixes #304

### CODE REVIEW

File: custom_components/tahoma/climate_devices/atlantic_electrical_heater.py [hunk 1]
```diff
@@ -1,43 +1,153 @@
...
+    """Representation of Atlantic Electrical Heater (With Adjustable Temperature Setpoint)."""
+
+    def __init__(self, device_url: str, coordinator: TahomaDataUpdateCoordinator):
+        """Init method."""
+        super().__init__(device_url, coordinator)
+
+        self._temp_sensor_entity_id = None
+        self._current_temperature = None
+
+    async def async_added_to_hass(self):
```
Comment (iMicknl) [inline custom_components/tahoma/climate_devices/atlantic_electrical_heater.py:89]: This is exactly the same code as @tetienne uses in SomfyThermostat. Should we put this in a separate class and inherit from this code.

Comment (tetienne) [reply custom_components/tahoma/climate_devices/atlantic_electrical_heater.py:89]: I would say yes. It will avoid to duplicate code and maintenance.

Comment (iMicknl) [reply custom_components/tahoma/climate_devices/atlantic_electrical_heater.py:89]: I am not 100% sure how to do this in the best way though.. It seems a bit complex with multiple inheritance and I am not able to test.. 
Let's do this in a follow up PR.

File: custom_components/tahoma/climate_devices/atlantic_electrical_heater.py [hunk 2]
```diff
@@ -18,26 +18,26 @@
 
 COMMAND_SET_HEATING_LEVEL = "setHeatingLevel"
 
+CORE_ON_OFF_STATE = "core:OnOffState"
 IO_TARGET_HEATING_LEVEL_STATE = "io:TargetHeatingLevelState"
 
-PRESET_FREEZE = "Freeze"
-PRESET_FROST_PROTECTION = "frostprotection"
-PRESET_OFF = "off"
+PRESET_FREEZE = "freeze"
```
Comment (tillstaff) [inline custom_components/tahoma/climate_devices/atlantic_electrical_heater.py:24]: Why did you rename "frost protection" to "freeze"? Frostprotection seems well named but I could be wrong.
In french, the function is named "hors-gel" and it will just set the temperature of the heater to 7°C to avoid any freezing in the house.

Although, the following presets are needed too in the Atlantic_Electrical_Heater device : 
PRESET_COMFORT1 = "comfort-1"
PRESET_COMFORT2 = "comfort-2"

File: custom_components/tahoma/climate_devices/atlantic_electrical_heater.py [hunk 3]
```diff
@@ -52,35 +52,31 @@ def supported_features(self) -> int:
...
     @property
     def hvac_modes(self) -> List[str]:
         """Return the list of available hvac operation modes."""
-        return [*MAP_HVAC_MODES]
+        return [*HVAC_MODES_TO_TAHOMA]
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        await self.async_execute_command(
+            COMMAND_SET_HEATING_LEVEL, HVAC_MODES_TO_TAHOMA[hvac_mode]
```
Comment (tillstaff) [inline custom_components/tahoma/climate_devices/atlantic_electrical_heater.py:65]: This command won't work for the PASS_ACTUATOR device.

Here are the commands sent by the Cozytouch application : 

Set to Off (the command is named "off" without parameters): 
```
2021-06-01 10:22:50 INFO (MainThread) [custom_components.tahoma] 2021-06-01 10:21:12: off executed via Android 2.5.4#244 Set heating level in room c423fb05-92c5-49b4-a917-c7ac553e7ef4 on io://xxxx-xxxx-xxxx/2634956, with [].
```

Set to On (there is no command for "on", it's just set the preset to "comfort"): 
```
2021-06-01 10:22:50 INFO (MainThread) [custom_components.tahoma] 2021-06-01 10:21:37: setHeatingLevel executed via Android 2.5.4#244 Set heating level in room c423fb05-92c5-49b4-a917-c7ac553e7ef4 on io://xxxx-xxxx-xxxx/2634956, with ['comfort'].
```

File: custom_components/tahoma/climate_devices/atlantic_electrical_heater.py (hunk 1 above)
Comment (iMicknl) [reply custom_components/tahoma/climate_devices/atlantic_electrical_heater.py]: Logged in https://github.com/iMicknl/ha-tahoma/issues/472.
